plugin.rb

```diff
@@ -40,4 +40,62 @@
   end
 
   DiscourseChatIntegration::Provider.mount_engines
+
+  if defined?(DiscourseAutomation)
+    add_automation_scriptable('send_slack_message') do
+      field :message, component: :message, required: true, accepts_placeholders: true
+      field :url, component: :text, required: true
+      field :channel, component: :text, required: true
+
+      version 1
+
+      triggerables %i[point_in_time recurring]
+
+      script do |context, fields, automation|
```

Comment:
testing automation can be slightly hard ATM due to how hard it is to test a conjunction of plugins.

One trick I like to do is exporting the core logic of you script into a lib. So basically in this case you could have some lib: `SendSlackMessageCreator.create(fields)` and ensure it generates what you expect in the different cases.

And then your script block would just be:


```ruby
script do |_, fields|
  SendSlackMessageCreator.create(fields)
end
```